content/networking.md

```diff
@@ -0,0 +1,190 @@
+wrong, like when the internet connection has dropped entirely.
+
+You might be seeing the word HTTP/2 getting thrown around regularly. You don't
+need to worry too much about this, as HTTP/2 doesn't change any existing browser
+APIs. You can think of it as a "more efficient version HTTP". For now, don't
+worry about things like HTTP/2 push either - they're mostly details for servers,
+which we won't be covering here.
+
+The browser has several ways of performing HTTP requests.
+- __Navigating to a new page:__ performs an HTTP request whenever you type in a
+  url in your browser. The server then replies with some `index.html` file which
+  contains more links to assets and other pages.
+- __Through `<script>`, `<link>` & other tags:__ When an `index.html` page is
+  loaded, the browser will read out all `<link>` and `<script>` tags in the
+  document's head. This will trigger requests for more resources, which in turn
+  can request even more resources. There's also the `<a>` and `<img>` tags,
+  which act similarly.
+- __Using `prefetch` headers.__ When a browser page is loaded, the server can
+  set headers for additional resources that should be loaded. The browser then
+  proceeds to request those.
+- __Using `new XMLHTTPRequest()` & `window.fetch()`:__ In order to make dynamic
+  requests to say, a JSON API, you can use these APIs. `XMLHTTPRequest` is the
+  classic way of performing requests, but these days there's the more modern
+  `window.fetch()` API. Regardless of which API you use, they produce similar
+  results.
+- __Using `navigator.sendBeacon()`:__ Sometimes you want to perform an HTTP
+  Request, but want to allow more important requests to be prioritized first.
+  For example with analytics data. The `sendBeacon()`  API, allows for exactly
+  this: it allows you to create an HTTP POST request, but schedules it to occur
+  in the background. Even if the page is closed before the request had a chance
+  to complete.
+- __Using (dynamic) `import`:__ Scripts can request other scripts, by using the
+  `import` syntax. This can either be dynamic or static - but in both cases it
+  makes an HTTP request to require JavaScript.
+
+### Fetch
+There's many ways of creating an HTTP request, but the most common one these
+days is using `fetch()`. In Choo you might want to trigger a request based on
+some other event coming through the EventEmitter. Let's look at a brief example
+of how to trigger a request based on a Choo event:
+
+_note: we recommend going through [the stores docs](/reference/stores) first. We
+assume you'll know how Choo's event emitter model works for the next few
+sections._
+
+```js
+var choo = require('choo')
+
+var app = choo()
+app.store((state, emitter) => {
+  state.tweets = []
+  emitter.on('fetch-tweets', (username) => {   // 1.
+    window.fetch(`/${username}/tweets`)        // 2.
+      .then((res) => res.json())               // 3.
+      .then((json) => JSON.parse(json))        // 4.
```

Comment:
`JSON.parse` doesn't return a promise, and fetchs `res.json()` returns a promise with the parsed json. so just delete this line ;)

Reply:
Oh, nice catch! - I haven't used the `fetch` API much haha 😅